src/network/electrum.rs

```diff
@@ -128,32 +127,187 @@ impl ElectrumConfig {
             url: electrum_url,
         }
     }
-    // Get a copy of the network (Chain) field.
-    pub fn network(&self) -> Chain {
+
+    pub fn build_bitcoin_client(&self) -> Result<ElectrumBitcoinClient, Error> {
+        ElectrumBitcoinClient::new(self.url.clone(), self.timeout, self.network)
+    }
+
+    pub fn build_liquid_client(&self) -> Result<ElectrumLiquidClient, Error> {
+        ElectrumLiquidClient::new(self.url.clone(), self.timeout, self.network)
+    }
+}
+
+pub struct ElectrumBitcoinClient {
+    inner: electrum_client::Client,
+    network: Chain,
+}
+
+impl ElectrumBitcoinClient {
+    fn new(url: ElectrumUrl, timeout: u8, network: Chain) -> Result<Self, Error> {
+        Ok(Self {
+            inner: url.build_client(timeout)?,
+            network,
+        })
+    }
+
+    fn fetch_utxos_core(
```

Comment:
NIT: fetch implies some IO like network request. Perhaps we can use `extract_script_utxos`?

Reply:
Change the name to `extract_address_utxos` in [58f93e5](https://github.com/SatoshiPortal/boltz-rust/pull/91/commits/58f93e5f6d974136f709e93b7fbebd4277f67c76)